### PR TITLE
Supportability enabled/disabled metrics

### LINKED ISSUE #2063: Wire up supportability metrics for security agent enablement configuration parameters
The `SecurityInterface` class should be updated to "record once" the following 2 supportability metrics the first time that the security agent enablement configuration parameters are checked.

The `Supportability/Ruby/SecurityAgent/Agent/Enabled/{enabled|disabled}` metric should be sent with an `enabled` / `disabled` value that corresponds to the `security.agent.enabled` configuration parameter's `true` / `false` value.

The `Supportability/Ruby/SecurityAgent/Enabled/{enabled|disabled}` metric should be sent with an `enabled` / `disabled` value that corresponds to the `security.enabled` configuration parameter's `true` / `false` value.

### PR DESCRIPTION
**Pull request by hannahramadan**
Add security agent enabled/disabled supportability metric

closes #2063

### CODE REVIEW

**Review by fallwith — APPROVED** [review]

**Review by kaylareopelle — CHANGES_REQUESTED** [review]
I think some of the test failures are related to this PR. This one is coming from Ruby 3.2.2 and rails61.

```
NewRelic::Control::SecurityInterfaceTest#test_initialization_short_circuits_if_the_agent_has_already_been_started [/home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/test/new_relic/control/security_interface_test.rb:74]:
Did not find stats for spec #<NewRelic::MetricSpec 'Supportability/Ruby/SecurityAgent/Agent/Enabled/enabled':''>.
All specs in there were: [
  Supportability/API/drop_buffered_data (<unscoped>),
  Supportability/API/manual_start (<unscoped>),
  Supportability/API/shutdown (<unscoped>)
]

rails test /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/test/new_relic/control/security_interface_test.rb:64
```

File: test/new_relic/control/security_interface_test.rb [hunk 1]
```diff
@@ -23,11 +31,15 @@ def test_initialization_short_circuits_when_the_security_agent_is_disabled
...
 
       refute_predicate NewRelic::Control::SecurityInterface.instance, :agent_started?
+      assert_metrics_recorded 'Supportability/Ruby/SecurityAgent/Agent/Enabled/disabled'
+      assert_metrics_recorded 'Supportability/Ruby/SecurityAgent/Enabled/enabled'
     end
     logger.verify
   end
 
   def test_initialization_short_circuits_when_the_security_is_disabled
+    reset_supportability_metrics
```
Comment (fallwith) [inline test/new_relic/control/security_interface_test.rb:41]: This is a nice helper method that will perform a commonly needed operation for multiple tests.

Perhaps a similar helper can be used for the 2 `assert_metrics_recorded` calls that are needed by so many tests.

File: test/new_relic/control/security_interface_test.rb [hunk 2]
```diff
@@ -13,7 +13,15 @@ def setup
     end
   end
 
+  # For testing purposes, clear out the supportability metrics that have already been recorded.
```
Comment (fallwith) [inline test/new_relic/control/security_interface_test.rb:16]: This comment doesn't seem to cover the need for the shutdown.

File: test/new_relic/control/security_interface_test.rb (hunk 1 above)
Comment (hannahramadan) [reply test/new_relic/control/security_interface_test.rb:45]: Great idea! There are a few places where both are enabled, and I've cleaned that up into a method https://github.com/newrelic/newrelic-ruby-agent/pull/2286/commits/da5b18cabce2aa041e459e8bb7d4f7322437ef94

File: test/new_relic/control/security_interface_test.rb (hunk 2 above)
Comment (hannahramadan) [reply test/new_relic/control/security_interface_test.rb:16]: Removed.

**Review by fallwith — APPROVED** [review]

**Review by kaylareopelle — APPROVED** [review]

**Pull request merged**